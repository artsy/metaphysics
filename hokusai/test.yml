version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
