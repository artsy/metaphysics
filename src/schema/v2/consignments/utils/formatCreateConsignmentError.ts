import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
