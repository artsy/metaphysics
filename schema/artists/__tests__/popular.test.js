import { runAuthenticatedQuery, runQuery } from "test/utils"
import gql from "test/gql"

describe("when logged in", () => {
  it("makes a call for popular artists to the auth'd popularArtistsLoader", () => {
    const query = gql`
      {
        popular_artists {
          artists {
            id
          }
        }
      }
    `

    const rootValue = {
      authenticatedPopularArtistsLoader: () =>
        Promise.resolve([
          { birthday: "1900", artworks_count: 100, id: "ortina" },
          { birthday: "1900", artworks_count: 100, id: "xtina" },
        ]),
    }

    expect.assertions(1)
    return runAuthenticatedQuery(query, rootValue).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})

describe("when an anonymous user", () => {
  it("makes a call for popular artists to the popularArtistsLoader", () => {
    const query = gql`
      {
        popular_artists {
          artists {
            id
          }
        }
      }
    `

    const rootValue = {
      popularArtistsLoader: () =>
        Promise.resolve([
          { birthday: "1900", artworks_count: 100, id: "ortina" },
          { birthday: "1900", artworks_count: 100, id: "xtina" },
        ]),
    }

    expect.assertions(1)
    return runQuery(query, rootValue).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})
