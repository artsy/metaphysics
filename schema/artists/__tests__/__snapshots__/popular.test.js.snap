// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when an anonymous user makes a call for popular artists to the popularArtistsLoader 1`] = `
Object {
  "popular_artists": Object {
    "artists": Array [
      Object {
        "id": "ortina",
      },
      Object {
        "id": "xtina",
      },
    ],
  },
}
`;

exports[`when logged in makes a call for popular artists to the auth'd popularArtistsLoader 1`] = `
Object {
  "popular_artists": Object {
    "artists": Array [
      Object {
        "id": "ortina",
      },
      Object {
        "id": "xtina",
      },
    ],
  },
}
`;
